src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -2314,7 +2314,12 @@ private static bool CheckIfMethodIsApplicableAndPrepareIt(MethodData method, Exp
 			if (method.MethodBase != null && method.MethodBase.IsGenericMethodDefinition &&
 			    method.MethodBase is MethodInfo)
 			{
-				var genericMethod = MakeGenericMethod(method);
+				MethodInfo genericMethod;
+#if NET7_0_OR_GREATER
+				genericMethod = TryMakeGenericMethod(method);
+#else
+				genericMethod = MakeGenericMethod(method);
+#endif
```

Comment:
You can simply replace `MakeGenericMethod` with `TryMakeGenericMethod`, and avoid a compilation target check.